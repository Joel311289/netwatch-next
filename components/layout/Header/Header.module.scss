.headerWrapper {
  @include box(100%, 100%);

  position: relative;
}

.logo {
  width: var(--size-header);
  height: var(--size-header);
  max-height: var(--size-header);
  content: var(--img-logo);
}

.title {
  @include font(32px, $color: var(--color-text));

  font-family: var(--font-logo);
  margin: 5px 15px 0;
}
