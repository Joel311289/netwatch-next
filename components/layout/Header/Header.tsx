import React from 'react';
import Link from 'next/link';
import Image from 'next/image';
import cn from 'classnames';
import { ElementChildren, ElementHTML } from '@/types';
import { Space } from '@/components/layout';
import styles from '@/components/layout/Header/Header.module.scss';

export type HeaderProps = typeof defaultProps &
  ElementHTML &
  ElementChildren<JSX.Element | Array<JSX.Element>> & {
    title: string;
    href?: string;
    logoUrl?: string;
  };

const defaultProps = {
  href: '',
  logoUrl: '/assets/images/logo-light.png'
};

const Header = ({ children, title, href, logoUrl }: HeaderProps) => {
  return (
    <Space justify="between" align="center" className={cn(styles.headerWrapper)}>
      <Link href={href}>
        <a role="link">
          <Space align="center">
            <Image className={styles.logo} src={logoUrl} alt="logo" width={58} height={58} />

            <p className={styles.title}>{title}</p>
          </Space>
        </a>
      </Link>

      <Space align="center" gap={10}>
        {children}
      </Space>
    </Space>
  );
};

Header.defaultProps = defaultProps;

export default Header;
