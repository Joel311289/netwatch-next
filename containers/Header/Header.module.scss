/** Container (main component) **/

.wrapper {
  @include box(100%, 100%);

  position: relative;
}

.logo {
  width: var(--size-header);
  height: var(--size-header);
  content: var(--img-logo);
}

.title {
  color: var(--color-text);
  font-family: var(--font-logo);
  margin: 5px 15px 0 10px !important;
}

/** HeaderActions (child component) **/

.headerAction {
  @include box(var(--size-header-action), var(--size-header-action) !important);

  border-color: var(--color-border) !important;
  min-width: var(--size-header-action);
  font-size: 20px !important;
  padding: 0 !important;
}
