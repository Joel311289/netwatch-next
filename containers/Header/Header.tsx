import Link from 'next/link';
import Image from 'next/image';
import { useTranslation } from 'next-i18next';
import { IconType } from 'react-icons';
import { FiSettings } from 'react-icons/fi';
import { ElementHTML } from '@/types';
import { Settings } from '@/containers/';
import { Space } from '@/components/layout';
import { Button } from '@/components/forms';
import { Heading } from '@/components/typography';
import { Drawer, Portal } from '@/components/overlay';
import { classes } from '@/utils/helpers';
import styles from '@/containers/Header/Header.module.scss';

/* -------------------------------------------------------------------------- */
/** HeaderActions (child component) **/
/* -------------------------------------------------------------------------- */

type HeaderActionsProps = {
  actions?: Array<{ label: string; Icon: IconType }>;
};

const defaultActions = [{ label: 'header.action.settings', Icon: FiSettings }];

const HeaderActions = ({ actions = defaultActions }: HeaderActionsProps) => {
  const { t } = useTranslation();

  return (
    <Space align="center" gap={20} className={styles.headerActions}>
      {actions.map(({ label, Icon }) => (
        <Portal key={label}>
          <Portal.Handler>
            <Button outline tooltip={t(label)} className={styles.headerAction}>
              <Icon />
            </Button>
          </Portal.Handler>
          <Portal.Paper>
            <Drawer position="right" heading={t(label)}>
              <Settings />
            </Drawer>
          </Portal.Paper>
        </Portal>
      ))}
    </Space>
  );
};

/* -------------------------------------------------------------------------- */
/** Container (main component) **/
/* -------------------------------------------------------------------------- */

export type HeaderProps = typeof defaultProps & ElementHTML;

const defaultProps = {};

const Header = ({}: HeaderProps) => {
  const { t } = useTranslation();

  return (
    <Space justify="between" align="center" className={classes(styles.wrapper)}>
      <Link href="/">
        <a>
          <Space align="center">
            <Image
              className={styles.logo}
              src="/assets/images/logo-light.png"
              alt="logo"
              width={58}
              height={58}
            />

            <Heading level={1} className={styles.title}>
              {t('application.name')}
            </Heading>
          </Space>
        </a>
      </Link>

      <Space align="center" gap={10}>
        <HeaderActions />
      </Space>
    </Space>
  );
};

Header.defaultProps = defaultProps;

export default Header;
