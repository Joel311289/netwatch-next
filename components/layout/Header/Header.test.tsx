import { render, screen } from '@testing-library/react';
import Header from '@/components/layout/Header/Header';

describe('Tests Header component', () => {
  test('renders component correctly', () => {
    const { container } = render(<Header title="Header" />);

    expect(container).toMatchSnapshot();
  });

  test('renders title correctly', () => {
    render(<Header title="Header" />);

    expect(screen.getByText('Header')).toBeInTheDocument();
  });

  test('renders logo image url correctly', () => {
    render(<Header title="Header" logoUrl="/assets/images/logo-light.png" />);

    expect(screen.getByAltText('logo')).toBeInTheDocument();
  });

  test('renders logo href link correctly', () => {
    render(<Header title="Header" href="/home" />);

    expect(screen.getByRole('link')).toHaveAttribute('href', '/home');
  });
});
