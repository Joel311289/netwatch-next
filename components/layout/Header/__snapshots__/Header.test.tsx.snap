// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Header component renders component correctly 1`] = `
<div>
  <div
    class="wrapper wrapper"
  >
    <a
      href="/"
    >
      <div
        class="wrapper"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2758%27%20height=%2758%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="logo"
            class="logo"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <p
          class="title"
        >
          Header
        </p>
        <style>
          
        .wrapper {
          display: flex;
          flex-direction: row;
          flex-wrap: nowrap;
          
          
          align-items: center
        }
      
        </style>
      </div>
    </a>
    <div
      class="wrapper"
    >
      <style>
        
        .wrapper {
          display: flex;
          flex-direction: row;
          flex-wrap: nowrap;
          gap: 10px
          
          align-items: center
        }
      
      </style>
    </div>
    <style>
      
        .wrapper {
          display: flex;
          flex-direction: row;
          flex-wrap: nowrap;
          
          justify-content: space-between
          align-items: center
        }
      
    </style>
  </div>
</div>
`;
